stripe-core/src/main/java/com/stripe/android/core/networking/DefaultAnalyticsRequestExecutor.kt

```diff
@@ -11,45 +21,139 @@ import kotlin.coroutines.CoroutineContext
 
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
 class DefaultAnalyticsRequestExecutor(
+    private val context: Context,
     private val stripeNetworkClient: StripeNetworkClient,
-    @IOContext private val workContext: CoroutineContext,
-    private val logger: Logger
+    private val workContext: CoroutineContext,
 ) : AnalyticsRequestExecutor {
 
-    constructor() : this(
-        Logger.noop(),
-        Dispatchers.IO
-    )
+    private val canUseWorkManager: Boolean by lazy {
+        val workManagerInClasspath = runCatching {
+            Class.forName("androidx.work.WorkManager")
+        }.isSuccess
+
+        workManagerInClasspath && WorkManager.isInitialized()
+    }
 
     @Inject
     constructor(
-        logger: Logger,
-        @IOContext workContext: CoroutineContext
+        context: Context,
+    ) : this(
+        context = context.applicationContext,
+        stripeNetworkClient = DefaultStripeNetworkClient(
+            workContext = Dispatchers.IO,
+            logger = Logger.getInstance(BuildConfig.DEBUG),
+        ),
+        workContext = Dispatchers.IO,
+    )
+
+    constructor(
+        context: Context,
+        workContext: CoroutineContext,
     ) : this(
-        DefaultStripeNetworkClient(
+        context = context.applicationContext,
+        stripeNetworkClient = DefaultStripeNetworkClient(
             workContext = workContext,
-            logger = logger
+            logger = Logger.getInstance(BuildConfig.DEBUG),
         ),
-        workContext,
-        logger
+        workContext = workContext,
     )
 
-    /**
-     * Make the request and ignore the response.
-     */
     override fun executeAsync(request: AnalyticsRequest) {
-        logger.info("Event: ${request.params[FIELD_EVENT]}")
+        if (canUseWorkManager) {
+            val workManager = WorkManager.getInstance(context)
+            enqueue(workManager, request)
+        } else {
+            execute(request)
+        }
+    }
+
+    private fun execute(request: AnalyticsRequest) {
+        val logger = Logger.getInstance(enableLogging = BuildConfig.DEBUG)
+        val eventName = request.params[FIELD_EVENT]
+        logger.info("Event: $eventName")
 
         CoroutineScope(workContext).launch {
             runCatching {
                 stripeNetworkClient.executeRequest(request)
+            }.onSuccess {
+                logger.info("Successfully sent event $eventName")
+            }.onFailure {
+                logger.error("Exception while sending event $eventName", it)
+            }
+        }
+    }
+
+    private fun enqueue(
```

Comment:
Thanks for adding this feature here! I think it is going to fix a [similar Jira](https://jira.corp.stripe.com/browse/IDPROD-4187) ticket tracking this for Identity, the difference is that Identity uses [AnalyticsRequestV2](https://github.com/stripe/stripe-android/blob/4fc1e6817ae31e6f3de4b8f7af92a42a9608b1e8/stripe-core/src/main/java/com/stripe/android/core/networking/AnalyticsRequestV2.kt#L34)(sends to `r.stripe.com` instead of `q.stripe.com`) and it is is [directly sent through](https://github.com/stripe/stripe-android/blob/4c337906e429f16273fb1de1c4086277030fb4af/identity/src/main/java/com/stripe/android/identity/networking/DefaultIdentityRepository.kt#L238) `StripeNetworkClient.execute()`.

I think the `WorkManager` impl could be beneficial to all `StripeRequest`s instead of just `AnalyticsRequest`, if you could move the logic into `DefaultStripeNetworkClient` we can share it for all requests,  wdyt?

Reply:
We shouldn’t use it for all `StripeRequest` instances. Most of our requests should be sent immediately and only once, whereas for analytics requests, we can delay them and retry them multiple times.

Reply:
I think even though it's only applicable to AnalyticsRequest now, extracting the WorkManager logic outside the requester would provide the flexibility to apply to other `StripeRequest`s should it's needed later.
That said extending `AnalyticsRequestExecutor` for V2 also works for now, will leave for you to decide.

Reply:
+1 to not using it on regular requests. If we find async requests useful on use cases different than analytics in the future, we can always create a dedicated request type (something like `AsyncRequest`)